src/simplnx/Utilities/ThreadedAlgorithm.hpp

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include "simplnx/Filter/IFilter.hpp"
+#include "simplnx/simplnx_export.hpp"
+
+#include <chrono>
+#include <cstdint>
+
+namespace nx::core
+{
+
+class SIMPLNX_EXPORT ThreadedAlgorithm
+{
+
+public:
+  ThreadedAlgorithm(const IFilter::MessageHandler& mesgHandler, const std::atomic_bool& shouldCancel);
+  virtual ~ThreadedAlgorithm() noexcept;
```

Comment:
Why is this dtor marked as virtual?

Reply:
The current implementation has algorithm classes inheriting from `ThreadedAlgorithm` since that keeps the same API has the algorithm classes that already had a version of the thread safe message function. This is also why the `setTotalElements()` function exists since the number of elements is calculated in the call operator not the constructor. In a future implementation, we could move away from inheritance.